### DISCUSSION TITLE
Push parsed dependencies into github's dependency api

### DISCUSSION THREAD

**Discussion by yetanotherion** [renovatebot/renovate#21215] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

The awesome [vulnerabilityAlert](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts) feature relies on Github's alerts[1]. 
Unfortunately, Github supports less [package managers](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems) than Renovate. Among others it does not support **gradle**.

What do you think about Renovate pushing the dependency graph it computed to [Github](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/using-the-dependency-submission-api)?
This would give access to the **vulnerabilityAlert** feature to Gradle projects hosted on Github. 

[1]: As stated in the [doc](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts), the feature depends on [Dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#enabling-the-dependency-graph) and [Dependabot alerts](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-security-and-analysis-settings-for-your-repository).

**Comment by rarkins** [comment]
Interesting idea. Unfortunately it would only work for a small subset of Renovate's additional package managers, right? i.e. those in https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems ?

**Reply by yetanotherion** [reply]
>Unfortunately it would only work for a small subset of Renovate's additional package managers, right?

AFAIU yes indeed!

**Comment by viceice** [comment]
We no longer need github to propose alerts[^1]

[^1]: https://docs.renovatebot.com/configuration-options/#osvvulnerabilityalerts

**Reply by yetanotherion** [reply]
>We no longer need github to propose alerts[1](https://github.com/renovatebot/renovate/discussions/21215#user-content-fn-1-b9e713ab4b99fce45d68facf1fb13039)

@viceice this is great! Let's assume I put `osvVulnerabilityAlerts` to true.

This would make the `vulnerabilityAlert` feature work the same way as it does with Github's alerts?

**Reply by yetanotherion** [reply]
Got my answer at
https://github.com/renovatebot/renovate/discussions/20542#discussioncomment-5208540
```
"osvVulnerabilityAlerts": true,
"vulnerabilityAlerts": {
  "enabled": true
}
```
this is great, thanks!

**Reply by rarkins** [reply]
A difference is that you won't get the alerts queryable within GitHub's interface

**Reply by yetanotherion** [reply]
Thanks @rarkins! I'll try it anyway. 
Just to be sure, are osvVulnerabililtyAlerts and dependabot alerts exclusive? 
Said otherwise, based on the value of the boolean, renovate will scan one or the other but not both?

**Reply by rarkins** [reply]
I'm not sure what happens if you enable both at the same time. There's nothing preventing us supporting both at once except technical work

**Reply by Churro** [reply]
Both GitHub and OSV alerts create package rules that require patched versions for packages that were found vulnerable. In case of overlap, renovate merges the security advisories in its PR description. They are styled differently, which may lead to confusion.

I was thinking about testing this co-existence more extensively. Probably, it would even be an option to list `vulnerabilityProviders` within the `vulnerabilityAlerts` object and make OSV the default. But that's just some ideas that would deserve more thoughts👍

**Reply by yetanotherion** [reply]
Thanks @rarkins and @Churro!
> it would even be an option to list vulnerabilityProviders within the vulnerabilityAlerts object and make OSV the default

This **vulnerabilityProviders** option would be **very adapted** to our situation. For us, the current ideal setup would be:
- use OSV for gradle/jvm repositories,
- use GitHub in python, npm repositories to benefit from the existing GitHub UI.
